cmd/awsS3Upload.go

```diff
@@ -0,0 +1,204 @@
+//     Otherwise, nil and an error from the call to PutObject
+func PutFile(c context.Context, api S3PutObjectAPI, input *s3.PutObjectInput) (*s3.PutObjectOutput, error) {
+	return api.PutObject(c, input)
+}
+
+func newAwsS3UploadUtils() awsS3UploadUtils {
+	utils := awsS3UploadUtilsBundle{
+		Command: &command.Command{},
+		Files:   &piperutils.Files{},
+	}
+	// Reroute command output to logging framework
+	utils.Stdout(log.Writer())
+	utils.Stderr(log.Writer())
+	return &utils
+}
+
+type AWSCredentials struct {
+	AWS_ACCESS_KEY_ID     string `json:"access_key_id"`
+	Bucket                string `json:"bucket"`
+	AWS_SECRET_ACCESS_KEY string `json:"secret_access_key"`
+	AWS_REGION            string `json:"region"`
+}
+
+func awsS3Upload(configOptions awsS3UploadOptions, telemetryData *telemetry.CustomData) {
+	// Utils can be used wherever the command.ExecRunner interface is expected.
+	// It can also be used for example as a mavenExecRunner.
+	utils := newAwsS3UploadUtils()
+
+	// For HTTP calls import  piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	// and use a  &piperhttp.Client{} in a custom system
+	// Example: step checkmarxExecuteScan.go
+
+	//Prepare Credentials
+	if configOptions.JSONCredentialsAWS == "" {
+		log.Entry().Fatal("AWS Credentials were not set!")
+	}
+
+	log.Entry().Infoln("Start reading AWS Credentials")
+	var obj AWSCredentials
+
+	err := json.Unmarshal([]byte(configOptions.JSONCredentialsAWS), &obj)
+	if err != nil {
+		log.Entry().
+			WithError(err).
+			Fatal("Could not read JSONCredentialsAWS")
+	}
+
+	//Set environment variables which are needed to initialize S3 Client
+	log.Entry().Infoln("Successfully read AWS Credentials. Setting up environment variables")
+	AWS_REGION_set := setenvIfEmpty("AWS_REGION", obj.AWS_REGION)
+	AWS_ACCESS_KEY_ID_set := setenvIfEmpty("AWS_ACCESS_KEY_ID", obj.AWS_ACCESS_KEY_ID)
+	AWS_SECRET_ACCESS_KEY_set := setenvIfEmpty("AWS_SECRET_ACCESS_KEY", obj.AWS_SECRET_ACCESS_KEY)
+
+	defer removeEnvIfPreviouslySet("AWS_REGION", AWS_REGION_set)
+	defer removeEnvIfPreviouslySet("AWS_ACCESS_KEY_ID", AWS_ACCESS_KEY_ID_set)
+	defer removeEnvIfPreviouslySet("AWS_SECRET_ACCESS_KEY", AWS_SECRET_ACCESS_KEY_set)
+
+	//Initialize S3 Client
+	log.Entry().Infoln("Loading Configuration for S3 Client")
+	cfg, err := config.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		log.Entry().
+			WithError(err).
+			Fatal("AWS Client Configuration failed")
+	}
+	client := s3.NewFromConfig(cfg)
+
+	// Error situations should be bubbled up until they reach the line below which will then stop execution
+	// through the log.Entry().Fatal() call leading to an os.Exit(1) in the end.
+	err = runAwsS3Upload(&configOptions, telemetryData, utils, client, obj.Bucket)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("Step execution failed")
+	}
+}
+
+func runAwsS3Upload(configOptions *awsS3UploadOptions, telemetryData *telemetry.CustomData, utils awsS3UploadUtils, client S3PutObjectAPI, bucket string) error {
```

Comment:
utils and telemetry are not used, can be removed. You can remove the whole generated `awsS3UploadUtils` if you don't need it.